SPECS/uefistored.spec

```diff
@@ -60,31 +60,47 @@ cp %{SOURCE1} %{buildroot}%{_datadir}/uefistored/
 # /var/lib/uefistored/ is used by secureboot-certs
 install -d %{buildroot}%{_localstatedir}/lib/uefistored/
 
-# varstored-tools expects /usr/share/varstored/ to hold these certificates
+# /usr/share/varstored is where XAPI writes the certificates when they have
+# been set for the pool - using the secureboot-certs script - and a VM starts,
+# and where varstored-tools expects them too.
+# For FHS compliance, we fill it with symlinks to files in /var/lib/uefistored
 install -d %{buildroot}%{_datadir}/varstored/
-ln -s %{_datadir}/uefistored/PK.auth %{buildroot}%{_datadir}/varstored/PK.auth
+ln -s %{_localstatedir}/lib/uefistored/PK.auth %{buildroot}%{_datadir}/varstored/PK.auth
 ln -s %{_localstatedir}/lib/uefistored/KEK.auth %{buildroot}%{_datadir}/varstored/KEK.auth
 ln -s %{_localstatedir}/lib/uefistored/db.auth %{buildroot}%{_datadir}/varstored/db.auth
 ln -s %{_localstatedir}/lib/uefistored/dbx.auth %{buildroot}%{_datadir}/varstored/dbx.auth
 
 %check
 make test
 
+%post
+# For compliance with UEFI specs, we want a PK to be available to uefistored
+# even if the pool has not been setup for Secure Boot using secureboot-certs.
+# One might believe that this scriptlet could cause issues for future host ISO upgrades,
+# after certificates have been loaded to XAPI using secureboot-certs. This PK may indeed
+# differ from the one in XAPI and thus not be appropriate.
+# But this issue solves itself automatically: the correct PK from XAPI db would
+# automatically overwrite the one on disk as soon as an UEFI VM starts.
+if [ ! -e /var/lib/uefistored/PK.auth ];
+then
+    cp -f /usr/share/uefistored/PK.auth /var/lib/uefistored/PK.auth
+fi
+
 %files
 %{_sbindir}/uefistored
 %{_sbindir}/varstored
 %dir %{_datadir}/uefistored
 %{_datadir}/uefistored/PK.auth
 %{_sbindir}/secureboot-certs
 %dir %{_localstatedir}/lib/uefistored
-
-%dir %{_datadir}/varstored
-%{_datadir}/varstored/KEK.auth
-%{_datadir}/varstored/PK.auth
-%{_datadir}/varstored/db.auth
-%{_datadir}/varstored/dbx.auth
+%{_datadir}/varstored
```

Comment:
Because I forgot to revert my change from when I tried to make it a symlink :o)

Reply:
Ah, so declaring a directory to imply "this directory and all of its contents" but not its files is not valid RPM?

Reply:
So, actually, I could have realized that `%{_datadir}/varstored` was enough, despite the fact that I really forgot to revert changes. The reason it would have worked is because `%{_datadir}/varstored` meant either "the varstored file" (when I turned it into a symlink in my WIP), or "the varstored directory and everything it contains".

On the other side, `%dir %{_datadir}/varstored` means "just the varstored dir, no implied recursive contents", that's why after this we have to declare each file one by one. I actually prefer this way because it is more explicit. If for some reason we break the creation of the symlinks in %install, `rpm` will detect the mismatch at the end of the build and fail... Which wouldn't happen if we simply kept `%{_datadir}/varstored`